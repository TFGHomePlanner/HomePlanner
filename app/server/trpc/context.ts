import * as trpc from "@trpc/server";
import { prisma } from "../../common/prisma";
import * as trpcExpress from "@trpc/server/adapters/express";

export async function createContext(
  ctx: trpcExpress.CreateExpressContextOptions
) {
  const { req, res } = ctx;
  return {
    req,
    res,
    prisma,
  };
}

export type Context = trpc.inferAsyncReturnType<typeof createContext>;
