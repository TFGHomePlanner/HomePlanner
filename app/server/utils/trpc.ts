import { createTRPCReact } from "@trpc/react-query";
import type { AppRouter } from "../trpc/router/_app";
import { httpBatchLink } from "@trpc/client";

export const trpc = createTRPCReact<AppRouter>({});