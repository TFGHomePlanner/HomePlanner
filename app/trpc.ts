import { createTRPCReact } from "@trpc/react-query";
import type { AppRouter } from "./server/trpc/router/_app";


export const trpc = createTRPCReact<AppRouter>();
