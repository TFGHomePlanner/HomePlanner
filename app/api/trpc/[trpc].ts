import * as trpcNext from "@trpc/server/adapters/next";
import { appRouter } from "../../server/trpc/router/_app";
import { createContext } from "../../server/trpc/context";

export default trpcNext.createNextApiHandler({
	router: appRouter,
	createContext,
});