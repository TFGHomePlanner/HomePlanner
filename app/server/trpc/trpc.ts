import { initTRPC } from "@trpc/server";
import { Context } from "./context";
/**
 * Inicializaci√≥n de la instancia tRPC en backend
 */
const t = initTRPC.context<Context>().create();

/**
 * Exporta el router reutilizable y helpers de ayuda
 * que pueden ser utilizados en todo el router.
 */
export const router = t.router;
export const publicProcedure = t.procedure;
