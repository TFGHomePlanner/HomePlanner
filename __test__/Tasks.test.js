import { trpc } from "../app/server/utils/trpc";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import UserProvider from "../app/context/userContext";
import TabHomeScreen from "../app/screens/(tabs)/home";
import { httpBatchLink } from "@trpc/client";
import { create } from "react-test-renderer";

test("renders correctly", () => {
	const queryClient = new QueryClient();
	const trpcClient = trpc.createClient({
		links: [
			httpBatchLink({
				url: "http://192.168.1.38:4000/trpc",
			}),
		],
	});

	const tree = create(
		<QueryClientProvider client={queryClient}>
			<trpc.Provider client={trpcClient} queryClient={queryClient}>
				<UserProvider>
					<TabHomeScreen />
				</UserProvider>
			</trpc.Provider>
		</QueryClientProvider>
	)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
