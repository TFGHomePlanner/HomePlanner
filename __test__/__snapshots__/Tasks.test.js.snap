// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8F3ED",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Inicio
  </Text>
  <View
    style={
      {
        "height": 1,
        "marginVertical": 30,
        "width": "80%",
      }
    }
  />
</View>
`;
